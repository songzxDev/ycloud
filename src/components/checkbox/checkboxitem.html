<label class="y-checkbox-item" data-bind="event:{click:hanldeClick},css:{'y-checkbox-item-checked': checked() && checked() != 0 }">
  <span class="y-checkbox-item--inner-warpper">
    <span class="y-checkbox-item-inner">
      <input type="checkbox" data-bind="attr:{disabled:disabled}">
    </span>
  </span>
  <span class="y-checkbox-label" data-bind="text:label"></span>
  <span class="y-checkbox-second-secondarylabel" data-bind="text:secondarylabel"></span>
</label>
