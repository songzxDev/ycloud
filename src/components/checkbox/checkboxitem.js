import template from './checkboxitem.html'
import ko from 'knockout'
function init (params) {
  this.label = params.label
  this.value = params.value
  this.checked = params.checked || ko.observable(false)
  params.valueList = params.valueList || []
  if (params.valueList.indexOf(this.value) >= 0) {
    this.checked(true)
  }
  params.valueList.subscribe && params.valueList.subscribe(val => {
    if (val.indexOf(this.value) >= 0) {
      this.checked(true)
    } else {
      this.checked(false)
    }
  })
  this.checked.subscribe(val => {
    let index = params.valueList.indexOf(this.value)
    if (index < 0 && val) {
      params.valueList.push(this.value)
    } else if (!val && index >= 0) {
      params.valueList.splice(index, 1)
    }
  })
  // checkbox重构，不和input进行绑定，在modal+checkbox-grid场景 全选始终无法选中，
  // 于是干脆不绑定了，直接用 试一下
  this.hanldeClick = (data, event) => {
    data.checked(!data.checked())
  }
}
export default {
  name: 'checkbox-item',
  init,
  template
}
