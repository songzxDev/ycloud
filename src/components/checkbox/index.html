<div class="y-checkbox" data-bind="foreach: {data: dataList, as: 'item'}, css:{'y-checkbox-hor':direction == 'horizontal','y-checkbox-ver':direction == 'vertical'}">
  <y-checkbox-item params="label: item.label, value: item.value, valueList: $parent.value"></y-checkbox-item>
</div>
