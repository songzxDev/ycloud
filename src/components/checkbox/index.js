import template from './index.html'
import ko from 'knockout'
import item from './checkboxitem'
const PREFIX = 'y-'
ko.components.register(PREFIX + item.name, {
  viewModel: item.init,
  template: item.template
})
function init (params) {
  if (params.dataList && params.dataList.subscribe) {
    this.dataList = params.dataList
  } else {
    this.dataList = ko.observableArray(params.dataList || [])
  }

  this.value = params.value || ko.observableArray([])
  this.valuebydot = params.valuebydot
  this.value.subscribe((val) => {
    if (this.valuebydot) {
      this.valuebydot(val.toString())
    }
  })
  this.direction = params.direction || 'horizontal'
}

export default {
  name: 'checkbox',
  init,
  template
}
