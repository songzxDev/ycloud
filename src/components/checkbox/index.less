.y-checkbox-ver .y-checkbox-item{
  display: block;
}
.y-cell .y-checkbox-item {
  margin-right: 10px;
}
.y-checkbox-item{
  cursor: pointer;
  font-size: @f-size-base;
  display: inline-block;
  margin-right: 25px;
  input{
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 1;
    cursor: pointer;
    opacity: 0;
  }
  .y-checkbox-label{
    position: relative;
    top: 1px;
  }
  .y-checkbox-second-secondarylabel{
    position: relative;
    top: 1px;
    color: @f-color-light
  }
}
.y-checkbox-item--inner-warpper{
  display: inline-block;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  outline: 0;
  line-height: 1;
  position: relative;
}
.y-checkbox-item-inner{
  display: inline-block;
  width: 14px;
  height: 14px;
  position: relative;
  top: 0;
  left: 0;
  border: 1px solid #dddee1;
  border-radius: 2px;
  background-color: #fff;
  transition: border-color .2s ease-in-out,background-color .2s ease-in-out;
}
.y-checkbox-item-checked .y-checkbox-item-inner{
  border-color: @primary-color;
  background-color: @primary-color;
  &:after{
    content: "";
    display: table;
    width: 4px;
    height: 8px;
    position: absolute;
    top: 1px;
    left: 4px;
    border-spacing: 0;
    border: 2px solid #fff;
    border-top: 0;
    border-left: 0;
    transform: rotate(45deg) scale(1);
    transition: all .2s ease-in-out;
  }
}
