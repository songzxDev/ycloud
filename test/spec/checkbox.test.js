import ko from 'knockout'
import checkbox from '../../src/components/checkbox/index.js'
import $ from 'jquery'
describe('y-checkbox组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + checkbox.name, {
    viewModel: checkbox.init,
    template: checkbox.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    title: ko.observable('test'),
    checkboxDataList: ko.observableArray([{value: 1,label:'男'}, {value: 0,label:'女'}]),
    checkboxValue: ko.observable(1)
  }
  document.body.innerHTML = '<y-checkbox params="dataList:checkboxDataList,value:checkboxValue"></y-checkbox>'
  ko.applyBindings(vm)
  var el = $('body')
  it("组件注册测试", () => {
    expect(checkbox).not.toBeNull();
    expect(ko.components.isRegistered(PREFIX + checkbox.name)).toBe(true);
  })
  it('生成checkbox测试', () => {
    expect($('body').find('.y-checkbox').length > 0).toEqual(true)
  })
  it('hor默认属性设置', () => {
    expect($('body').find('.y-checkbox-hor').length > 0).toEqual(true)
  })
  // todo radio方向测试
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
