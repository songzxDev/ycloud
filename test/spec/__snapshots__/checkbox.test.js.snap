// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-checkbox组件 快照 1`] = `
<body>
  <y-checkbox
    params="dataList:checkboxDataList,value:checkboxValue"
  >
    <div
      class="y-checkbox y-checkbox-hor"
      data-bind="foreach: {data: dataList, as: 'item'}, css:{'y-checkbox-hor':direction == 'horizontal','y-checkbox-ver':direction == 'vertical'}"
    >
      
  
      <y-checkbox-item
        params="label: item.label, value: item.value, valueList: $parent.value"
      />
      

      
  
      <y-checkbox-item
        params="label: item.label, value: item.value, valueList: $parent.value"
      />
      

    </div>
    

  </y-checkbox>
</body>
`;
